lib/net.c

```diff
@@ -42,17 +43,35 @@
 
 static struct net_config config;
 
-/*
- * XXX The secret key of the RSS hash must be random
- * in order to avoid hackers to know it.
- */
-uint8_t default_rss_key[GATEKEEPER_RSS_KEY_LEN] = {
-	0x6d, 0x5a, 0x56, 0xda, 0x25, 0x5b, 0x0e, 0xc2,
-	0x41, 0x67, 0x25, 0x3d, 0x43, 0xa3, 0x8f, 0xb0,
-	0xd0, 0xca, 0x2b, 0xcb, 0xae, 0x7b, 0x30, 0xb4,
-	0x77, 0xcb, 0x2d, 0xa3, 0x80, 0x30, 0xf2, 0x0c,
-	0x6a, 0x42, 0xb7, 0x3b, 0xbe, 0xac, 0x01, 0xfa,
-};
+uint8_t default_rss_key[GATEKEEPER_RSS_KEY_LEN];
+
+/* Function to randomize the default RSS key. */
+static void
+randomize_rss_key(void)
+{
+	uint8_t i, temp_key, temp_count;
+	uint8_t final_set_count = 0;
+	ssize_t number_of_bytes;
+
+	number_of_bytes = getrandom(default_rss_key, GATEKEEPER_RSS_KEY_LEN, 0);
+	if (number_of_bytes != 40) {
+		randomize_rss_key();
+	}
+
+	/* Count the number of bits set to 1 using Brian Kernighan's method. */
+	for (i = 0; i < GATEKEEPER_RSS_KEY_LEN; i++) {
+		temp_key = default_rss_key[i]; 
+		for (temp_count = 0; temp_key; temp_count++)
+		{
+	  		temp_key &= temp_key - 1;
+		}
+		final_set_count += temp_count;
+	}
+
+	if (final_set_count < 32 || final_set_count > 288) {
+		randomize_rss_key();
```

Comment:
Here, I would use a ``while`` loop instead of the recursive call. That is, ``while (final_set_count < min_num_set_bits || final_set_count > max_num_set_bits) {}``, please break it into lines as needed.